sqlglot/expressions.py

```diff
@@ -3150,7 +3150,7 @@ def alias_(expression, alias, table=False, dialect=None, quoted=None, **opts):
     alias = to_identifier(alias, quoted=quoted)
     alias = TableAlias(this=alias) if table else alias
 
-    if "alias" in exp.arg_types:
+    if "alias" in exp.arg_types and not isinstance(exp, Window):
```

Comment:
What if we switch to an allowlist instead of a denylist?

Reply:
Deny list seemed like a better choice for this one because the builder for all other expression types (eg. lateral, values, etc) that support the "alias" argument behaves as expected here.

Reply:
K. Then let's add a comment here about why the `alias` arg of Window shouldn't be used.

Reply:
I'll make a note to add that as an extra in my next PR!